internal/bucket/lifecycle/lifecycle.go

```diff
@@ -401,6 +385,17 @@ func (lc Lifecycle) eval(obj ObjectOpts, now time.Time) Event {
 			}
 		}
 
+		// DelMarkerExpiration
+		if obj.IsLatest && obj.DeleteMarker && !rule.DelMarkerExpiration.Empty() {
+			if due, ok := rule.DelMarkerExpiration.NextDue(obj); ok && (now.IsZero() || now.After(due)) {
+				return Event{
```

Comment:
Returning here can confuse the ILM prediction header, it is when now.IsZero() is true, we can have transition + expiration and expiration due time is way after the transition due time. I think the spirit of this code is that we evaluate all rules and then we order them later.

Reply:
Ah yes, good catch. I overlooked this.

Reply:
@vadmeste I have add a couple of unit tests to cover this.